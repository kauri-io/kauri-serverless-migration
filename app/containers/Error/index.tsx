import styled from 'styled-components';
import PrimaryButton from "../../components/Button/PrimaryButton";
import Link from 'next/link'

const ErrorContainer = styled.div`
    background: ${props => props.theme.bgPrimary};
    color: white;
    display: flex;
    align-items: center;
    justify-content: center;
    flex: 1;
    height: calc(100vh - 133px);
    flex-direction: column;
`;

// Because Nelson
const Code = styled.div`
    font-size: 90px;
`;

const Copy = styled.div`
    font-size: ${props => props.theme.fontSizes[3]}px;
    margin-bottom: ${props => props.theme.space[3]}px;
`;

const Error = () => <ErrorContainer>
    <Code>404</Code>
    <Copy>Sorry, something went wrong :(</Copy>
    <Link href="/"><PrimaryButton>Go to homepage</PrimaryButton></Link>
</ErrorContainer>

export default Error