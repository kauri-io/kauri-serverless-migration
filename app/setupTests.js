import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
