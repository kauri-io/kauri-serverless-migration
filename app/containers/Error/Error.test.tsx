import Error from './index'
import { mountWithTheme } from '../../setupTests'

describe('containers/Error', () => {
    it('snapshot should match', () => {
        const props = {}
        const wrapper = mountWithTheme(<Error {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
