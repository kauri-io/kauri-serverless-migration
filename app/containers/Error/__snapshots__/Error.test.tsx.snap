// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Error should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "DEFAULT_CARD_WIDTH": 288,
      "MAIN_BODY_WIDTH": 1242,
      "bg": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "bgPrimary": "#1E2428",
      "bgSecondary": "#1E3D3B",
      "borderTextArea": "#979797",
      "breakpoints": Array [
        "500px",
        "52em",
        "1242px",
      ],
      "colors": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
      "disabledBackgroundColor": "#D6D6D6",
      "disabledTextColor": "#9B9B9B",
      "divider": "#EBEBEB",
      "errorRedColor": "#C03030",
      "fontSizes": Array [
        11,
        13,
        14,
        16,
        18,
        20,
        22,
        28,
        32,
      ],
      "fontWeight": Array [
        300,
        "normal",
        500,
        700,
      ],
      "hoverTextColor": "#108E72",
      "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
      "padding": "calc((100vw - 1242px) / 2)",
      "paddingTop": "2.5em",
      "primary": "#0BA986",
      "primaryColor": "#0BA986",
      "primaryDark": "#11856B",
      "primaryTextColor": "#1E2428",
      "secondary": "#11856B",
      "secondaryBlue": "rgb(81, 123, 162, 1)",
      "secondaryBlueDark": "rgb(81, 118, 142, 1)",
      "secondaryColor": "#1E3D3B",
      "secondaryTextColor": "#283035",
      "space": Array [
        0,
        10,
        15,
        30,
        60,
      ],
      "tertiaryBackgroundColor": "#F7F7F7",
      "textPrimary": "#1E2428",
      "white": "#FFFFFF",
    }
  }
>
  <Error>
    <Error__ErrorContainer>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "Error__ErrorContainer-sc-9zpcuf-0",
              "isStatic": false,
              "lastClassName": "ieAeQL",
              "rules": Array [
                "background:",
                [Function],
                ";color:white;display:flex;align-items:center;justify-content:center;flex:1;height:calc(100vh - 133px);flex-direction:column;",
              ],
            },
            "displayName": "Error__ErrorContainer",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "Error__ErrorContainer-sc-9zpcuf-0",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="Error__ErrorContainer-sc-9zpcuf-0 ieAeQL"
        >
          <Error__Code>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "Error__Code-sc-9zpcuf-1",
                    "isStatic": true,
                    "lastClassName": "gGjEEd",
                    "rules": Array [
                      "font-size:90px;",
                    ],
                  },
                  "displayName": "Error__Code",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "Error__Code-sc-9zpcuf-1",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="Error__Code-sc-9zpcuf-1 gGjEEd"
              >
                404
              </div>
            </StyledComponent>
          </Error__Code>
          <Error__Copy>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "Error__Copy-sc-9zpcuf-2",
                    "isStatic": false,
                    "lastClassName": "hMEIwS",
                    "rules": Array [
                      "font-size:",
                      [Function],
                      "px;margin-bottom:",
                      [Function],
                      "px;",
                    ],
                  },
                  "displayName": "Error__Copy",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "Error__Copy-sc-9zpcuf-2",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="Error__Copy-sc-9zpcuf-2 hMEIwS"
              >
                Sorry, something went wrong :(
              </div>
            </StyledComponent>
          </Error__Copy>
          <Link
            href="/"
            prefetch={true}
          >
            <PrimaryButtonComponent
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <PrimaryButton
                bg="primary"
                color="white"
                fontSize={0}
                fontWeight={700}
                mr={2}
                onClick={[Function]}
                type="submit"
              >
                <StyledComponent
                  bg="primary"
                  color="white"
                  fontSize={0}
                  fontWeight={700}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "PrimaryButton-sc-1cbxa53-0",
                        "isStatic": false,
                        "lastClassName": "gGcsYX",
                        "rules": Array [
                          "display:flex;align-items:center;justify-content:center;min-width:136px;max-width:150px;",
                          [Function],
                          ";height:40px;border-radius:4px;border:none;cursor:pointer;> svg,img{height:16px;",
                          [Function],
                          ";}text-transform:uppercase;opacity:",
                          [Function],
                          ";",
                          [Function],
                          ";",
                          [Function],
                          ";",
                          [Function],
                          ";",
                          [Function],
                          ";:hover{cursor:",
                          [Function],
                          ";}",
                          ";:hover{background-color:",
                          [Function],
                          ";}",
                        ],
                      },
                      "displayName": "PrimaryButton",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "PrimaryButton-sc-1cbxa53-0",
                      "target": "button",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  mr={2}
                  onClick={[Function]}
                  type="submit"
                >
                  <button
                    className="PrimaryButton-sc-1cbxa53-0 gGcsYX"
                    color="white"
                    fontSize={0}
                    fontWeight={700}
                    onClick={[Function]}
                    type="submit"
                  >
                    Go to homepage
                  </button>
                </StyledComponent>
              </PrimaryButton>
            </PrimaryButtonComponent>
          </Link>
        </div>
      </StyledComponent>
    </Error__ErrorContainer>
  </Error>
</ThemeProvider>
`;
